src/components/Contacts/ContactFilters/ContactFilters.tsx

```diff
@@ -1,35 +1,211 @@
+                <ListItem data-testid="ErrorState">
+                  <ListItemText
+                    primary={error.toString()}
+                    primaryTypographyProps={{
+                      variant: 'subtitle1',
+                      color: 'error',
+                    }}
+                  />
+                </ListItem>
+              )}
+              {loading ? (
+                <ListItem data-testid="LoadingState">
+                  <CircularProgress />
+                </ListItem>
+              ) : data?.contactFilters?.length == 0 ? (
+                <ListItem data-testid="NoFiltersState">
+                  <ListItemText
+                    primary={t('No Contact Filters Found')}
+                    primaryTypographyProps={{ variant: 'subtitle1' }}
+                  />
+                </ListItem>
+              ) : (
+                <>
+                  {data?.contactFilters?.map((group: ContactFilterGroup) => (
+                    <Collapse
+                      key={group.id}
+                      in={showAll || isGroupVisible(group)}
+                      data-testid="FilterGroup"
+                    >
+                      <ListItem button onClick={() => showGroup(group)}>
+                        <ListItemText
+                          primary={group.title}
+                          primaryTypographyProps={{ variant: 'subtitle1' }}
+                        />
+                        <ArrowForwardIos fontSize="small" color="disabled" />
+                      </ListItem>
+                    </Collapse>
+                  ))}
+                  {data?.contactFilters?.some(
+                    (g: ContactFilterGroup) => !isGroupVisible(g),
+                  ) == true ? (
+                    <FilterListItemShowAll
+                      showAll={showAll}
+                      onToggle={() => setShowAll(!showAll)}
+                    />
+                  ) : null}
+                </>
+              )}
+            </FilterList>
+          </div>
+        </Slide>
+        <Slide
+          in={selectedGroup != null}
+          direction="left"
+          mountOnEnter
+          unmountOnExit
+        >
+          <div>
+            <FilterHeader>
+              <IconButton
+                size="small"
+                edge="start"
+                onClick={() => showGroup(null)}
+                style={{ verticalAlign: 'middle' }}
+              >
+                <ArrowBackIos fontSize="small" />
+              </IconButton>
+              <Typography
+                variant="h6"
+                component="span"
+                style={{ verticalAlign: 'middle' }}
+              >
+                {selectedGroup?.title}
+              </Typography>
+            </FilterHeader>
+            <FilterList dense>
+              {selectedGroup?.filters?.map((filter) => (
+                <FilterListItem
+                  key={filter.id}
+                  filter={filter as Filter}
```

Comment:
We should be able to make the child components use the updated types from `Filter.ts`:
```suggestion
                  filter={filter}
```